packages/frontend/editor-ui/src/composables/useWorkflowExtraction.ts

```diff
@@ -0,0 +1,549 @@
+		selectionVariables: Map<string, string>,
+		selectionChildrenVariables: Map<string, string>,
+		startNodeName: string,
+		returnNodeName: string,
+	): IWorkflowDataCreate {
+		const newConnections = Object.fromEntries(
+			Object.entries(connections).filter(([k]) => nodes.some((x) => x.name === k)),
+		);
+		if (end) {
+			// this is necessary because the new workflow may crash looking for the
+			// nodes in these connections
+			delete newConnections[end];
+		}
+
+		const startNodeTarget = nodes.find((x) => x.name === start);
+		const firstNode = startNodeTarget ?? nodes.sort((a, b) => a.position[1] - b.position[1])[0];
+		const startNodePosition: [number, number] = [
+			firstNode.position[0] - PUSH_NODES_OFFSET,
+			firstNode.position[1],
+		];
+
+		const endNodeTarget = nodes.find((x) => x.name === end);
+		const lastNode = endNodeTarget ?? nodes.sort((a, b) => b.position[1] - a.position[1])[0];
+		const endNodePosition: [number, number] = [
+			lastNode.position[0] + PUSH_NODES_OFFSET,
+			lastNode.position[1],
+		];
+
+		const shouldInsertReturnNode = selectionChildrenVariables.size > 0;
+
+		const startNodeConnection = startNodeTarget
+			? ({
+					[startNodeName]: {
+						main: [
+							[
+								{
+									node: startNodeTarget.name,
+									type: 'main',
+									index: 0,
+								},
+							],
+						],
+					},
+				} satisfies IConnections)
+			: {};
+
+		const endNodeConnection =
+			endNodeTarget && shouldInsertReturnNode
+				? ({
+						[endNodeTarget.name]: {
+							main: [
+								[
+									{
+										node: returnNodeName,
+										type: 'main',
+										index: 0,
+									},
+								],
+							],
+						},
+					} satisfies IConnections)
+				: {};
+
+		const returnNode = shouldInsertReturnNode
+			? [
+					{
+						parameters: {
+							assignments: {
+								assignments: [
+									...selectionChildrenVariables.entries().map((x) => ({
+										id: uuidv4(),
+										name: x[0],
+										value: `={{ ${x[1]} }}`,
+										type: 'string',
+									})),
+								],
+							},
+							options: {},
+						},
+						type: 'n8n-nodes-base.set',
+						typeVersion: 3.4,
+						position: endNodePosition,
+						id: uuidv4(),
+						name: returnNodeName,
+					} satisfies INode,
+				]
+			: [];
+		const triggerParameters =
+			selectionVariables.size > 0
+				? {
+						workflowInputs: {
+							values: [...selectionVariables.keys().map((k) => ({ name: k, type: 'any' }))],
+						},
+					}
+				: {
+						inputSource: 'passthrough',
+					};
+
+		const triggerNode: INode = {
+			id: uuidv4(),
+			typeVersion: 1.1,
+			name: startNodeName,
+			type: EXECUTE_WORKFLOW_TRIGGER_NODE_TYPE,
+			position: startNodePosition,
+			parameters: triggerParameters,
+		};
+
+		return {
+			name: newWorkflowName,
+			nodes: [...nodes, ...returnNode, triggerNode],
+			connections: {
+				...newConnections,
+				...startNodeConnection,
+				...endNodeConnection,
+			},
+			settings: { executionOrder: 'v1' },
```

Comment:
Should we refuse to do the extraction if the workflow we're extracting from is some legacy one not using `v1`?

Reply:
I have no idea how often those really exist in the wild, but I do know that they are executed slightly differently on the execution engine - if we extract nodes from them into sub workflows is there a risk something changes? 🤔 

Then again this might be an edge case we can happily scope out of the MVP